implementation/src/main/java/io/smallrye/config/AbstractLocationConfigSourceFactory.java

```diff
@@ -0,0 +1,237 @@
+                    addConfigSource(resourceUrl, configSources);
+                }
+            }
+
+        } catch (IOException e) {
+            throw ConfigMessages.msg.failedToLoadResource(e);
+        }
+    }
+
+    private void tryHttpResource(final URL url, final List<ConfigSource> configSources, final ConfigSourceContext context) {
+        if (url.getProtocol().startsWith("http")) {
+            if (validExtension(url.toString())) {
+                addConfigSource(url, configSources);
+                tryProfiles(url, configSources, context);
+            }
+        }
+    }
+
+    private void tryProfiles(final URL url, final List<ConfigSource> configSources, final ConfigSourceContext context) {
+        final List<String> profiles = context.getProfiles();
+        for (int i = profiles.size() - 1; i >= 0; i--) {
+            final int lastOrdinal = configSources.get(configSources.size() - 1).getOrdinal();
+            final URL profileToFileName = addProfileToFileName(url, profiles.get(i));
+            consumeAsPath(profileToFileName,
+                    path -> addProfileConfigSource(toURL(path.toUri()), lastOrdinal + 1, configSources));
+        }
+    }
+
+    private static URL toURL(final URI uri) {
+        try {
+            return uri.toURL();
+        } catch (MalformedURLException e) {
+            throw new IllegalArgumentException(e);
+        }
+    }
+
+    private void addConfigSource(final URL url, final List<ConfigSource> configSources) {
+        try {
+            final ConfigSource configSource = loadConfigSource(url);
+            if (!configSource.getProperties().isEmpty()) {
+                configSources.add(configSource);
+            }
+        } catch (IOException e) {
+            throw ConfigMessages.msg.failedToLoadResource(e);
+        }
+    }
+
+    private void addProfileConfigSource(final URL url, final int ordinal, final List<ConfigSource> configSources) {
+        try {
+            final ConfigSource configSource = loadConfigSource(url, ordinal);
+            if (!configSource.getProperties().isEmpty()) {
+                configSources.add(configSource);
+            }
+        } catch (FileNotFoundException e) {
+            // It is ok to not find the resource here, because it is an optional profile resource.
+        } catch (IOException e) {
+            throw ConfigMessages.msg.failedToLoadResource(e);
+        }
+    }
+
+    private boolean validExtension(final String resourceName) {
+        for (String s : getFileExtensions()) {
+            if (resourceName.endsWith(s)) {
+                return true;
+            }
+        }
+        return false;
+    }
+
+    private static URL addProfileToFileName(final URL fileName, final String profile) {
+        final int dot = fileName.toString().lastIndexOf(".");
+        try {
+            if (dot != -1) {
+                return new URL(fileName.toString().substring(0, dot) + "-" + profile + fileName.toString().substring(dot));
```

Comment:
Building URLs like this is strongly discouraged (think "URL injection").  This is another reason to prefer URI: it will ensure that the encoding of the URL is done correctly, something that is definitely not done here.